contracts/DKVDaggerHashimoto.sol

```diff
@@ -291,18 +309,19 @@ contract DecentralizedKVDaggerHashimoto is DecentralizedKV {
             uint256 chunkIdx = parent + (startShardId << (shardEntryBits + chunkLenBits));
             uint256 kvIdx = chunkIdx >> chunkLenBits;
             chunkIdx = chunkIdx % (1 << chunkLenBits);
-            bytes memory data = systemContract.unmaskWithEthash(kvIdx, maskedData[i]);
+            PhyAddr memory kvInfo = kvMap[idxMap[kvIdx]];
+            bytes memory unmaskedData = systemContract.unmaskChunkWithEthash(uint64(chunkIdx), kvInfo.hash, miner, maskedData[i]);
```

Comment:
Okay here be careful, we kvInfo.hash itself is in bytes 24. But the corresponding parameter in function sign of unmaskChunkWithEthash  is  in bytes 32. Automatical convertion occurs here!

Reply:
An explicit comments might be a good idea. Or directly change unmaskedChunkEthash's parameter from 32 bytes to 24 bytes.

Reply:
very constructive comments for me .